### PR TITLE
[GCP] Migrate compute metrics dashboard to lens

### PR DESCRIPTION
**Pull request by tommyers-elastic**
## What does this PR do?

Migrates GCP compute metrics dashboard visualizations to lens. Adds top-level datastream filter to improve performance.

**Note to reviewers:** when migrating the 'Read/Write IO' and "Network sent/received' panels, I noticed that the existing aggregations are based on doc count, not top values of the metric. I thought this was a bit strange. The resulting charts are quite different, as you can see in the screenshots below. One additional change in the new panels, is that I have left the 'Other' groupings in the 'top values' aggs, which can be removed if deemed necessary. 

Before: 

Screenshot 2023-01-18 at 12 13 40
Screenshot 2023-01-18 at 12 13 46

After:

Screenshot 2023-01-18 at 12 13 55
Screenshot 2023-01-18 at 12 14 02

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] Install and validate the new dashboard locally.

## How to test this PR locally

```
cd packages/gcp
elastic-package build .
elastic-package stack up -d --services=package-registry
...
```

## Related issues

Tracked by #5015.

### CODE REVIEW

**Comment by endorama** [comment]
I installed the package in this PR but the dashboard does not seem to be updated, I still see the "Before" version. @tommyers-elastic do you have any suggestion? I started from scratch with a 8.6.0 stack started through elastic-package.

File: packages/gcp/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,14 @@
 # newer versions go on top
+- version: "2.17.0"
+  changes:
+    - description: Migrate compute dashboard to lens and add datastream filter.
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/5038
+- version: "2.16.1"
```
Comment (endorama) [inline packages/gcp/changelog.yml:7]: Is this a partial leftover of https://github.com/elastic/integrations/pull/5032? I noticed mergify does complain about the head SHA, maybe we should merge than one before?

**Review by endorama — DISMISSED** [review]
Overall the dashboard looks good and works as expected. 2 minor comments:
- it seems this PR builds upon https://github.com/elastic/integrations/pull/5032
- tags (that have been exported through elastic-package) should not be added

File: packages/gcp/changelog.yml (hunk 1 above)
Comment (tommyers-elastic) [reply packages/gcp/changelog.yml:7]: yeh i just branched from 5032.

i will fix the inclusion of the tags over there and rebase

**Review by gpop63 — APPROVED** [review]
I tested the dashboard with data, the data stream filter is applied, seems fine to me!

**Review by endorama — APPROVED** [review]
comments address, lets get to main!! :shipit:

**Pull request merged**